mlflow/utils/process.py

```diff
@@ -92,6 +92,7 @@ def _exec_cmd(
             # https://docs.python.org/3/library/subprocess.html#subprocess.run
             kwargs["stderr"] = subprocess.STDOUT
 
+    input_str = kwargs.pop("input_str", None)
```

Comment:
I made the following change:

```diff
diff --git a/mlflow/pyfunc/backend.py b/mlflow/pyfunc/backend.py
index 429170f6c..6219e7b4e 100644
--- a/mlflow/pyfunc/backend.py
+++ b/mlflow/pyfunc/backend.py
@@ -159,14 +159,7 @@ class PyFuncBackend(FlavorBackend):
             if output_path:
                 predict_cmd += ["--output-path", str(output_path)]
 
-            input_str = sys.stdin.read() if input_path is None else None
-
-            return self.prepare_env(local_path).execute(
-                " ".join(predict_cmd),
-                stdin=subprocess.PIPE,
-                stdout=subprocess.PIPE,
-                input_str=input_str,
-            )
+            return self.prepare_env(local_path).execute(" ".join(predict_cmd))
         else:
             scoring_server._predict(local_uri, input_path, output_path, content_type)
 
diff --git a/mlflow/utils/environment.py b/mlflow/utils/environment.py
index e68f265b6..06239c181 100644
--- a/mlflow/utils/environment.py
+++ b/mlflow/utils/environment.py
@@ -594,7 +594,6 @@ class Environment:
         stderr=None,
         stdin=None,
         synchronous=True,
-        input_str=None,
     ):
         if command_env is None:
             command_env = os.environ.copy()
@@ -623,5 +622,4 @@ class Environment:
             stdout=stdout,
             stderr=stderr,
             stdin=stdin,
-            input_str=input_str,
         )
diff --git a/mlflow/utils/process.py b/mlflow/utils/process.py
index 9836c637c..a27d36a40 100644
--- a/mlflow/utils/process.py
+++ b/mlflow/utils/process.py
@@ -92,7 +92,6 @@ def _exec_cmd(
             # https://docs.python.org/3/library/subprocess.html#subprocess.run
             kwargs["stderr"] = subprocess.STDOUT
 
-    input_str = kwargs.pop("input_str", None)
     process = subprocess.Popen(
         cmd,
         env=env,
@@ -106,10 +105,8 @@ def _exec_cmd(
         for output_char in iter(lambda: process.stdout.read(1), ""):
             sys.stdout.write(output_char)
 
-    stdout, stderr = process.communicate(input=input_str)
+    stdout, stderr = process.communicate()
     returncode = process.poll()
-    if input_str and stdout and process.stdout:
-        sys.stdout.write(stdout)
     comp_process = subprocess.CompletedProcess(
         process.args,
         returncode=returncode,
```

and `pytest tests/models/test_cli.py::test_predict -s` passed. Can we comment out stdout-stdin-related changes and run `test_predict`?